src/main/java/com/dell/pravega/perf/PerfStats.java

```diff
@@ -0,0 +1,347 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.dell.pravega.perf;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.locks.LockSupport;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+
+/**
+ * class for Performance statistics.
+ */
+public class PerfStats {
+    final private String action;
+    final private String csvFile;
+    final private int messageSize;
+    final private long windowInterval;
+    private TimeWindow window;
+    final private ConcurrentLinkedQueue<TimeStamp> queue;
+    final private ForkJoinPool executor;
+    private LatencyWriter latencyRecorder;
+
+
+    /**
+     * private class for start and end time.
+     */
+    private class TimeStamp {
+        final private long start;
+        final private long end;
+        final private int bytes;
+
+        TimeStamp(long start, long end, int bytes) {
+            this.start = start;
+            this.end = end;
+            this.bytes = bytes;
+        }
+    }
+
+    /**
+     * private class for Performance statistics within a given time window.
+     */
+    private class TimeWindow {
```

Comment:
Should this be annotated NotThreadSafe?

Reply:
do you need NotThreadSafe for private class and its not used by multiple threads too.